app/js/components/service_status.ts

```diff
@@ -0,0 +1,156 @@
+import * as Chart from 'chart.js';
+import {ChartTooltipItem} from "chart.js";
+
+interface StatesDictionary {
+  [key: string]: string;
+}
+
+interface EntityJson {
+  name: string,
+  environment: string,
+  link: string,
+}
+
+interface StatesResponseJson {
+  service: {
+    entities: EntityJson,
+    link: string,
+    name: string,
+    states: StatesDictionary,
+    labels: StatesDictionary,
+    tooltips: StatesDictionary,
+  }
+}
+
+class ServiceStatus {
+  private stateColors: StatesDictionary;
+  private context: CanvasRenderingContext2D;
+  private serviceId: string;
+  private chart: Chart|undefined;
+
+  constructor(context: CanvasRenderingContext2D, serviceId: string) {
+    this.context = context;
+    this.serviceId = serviceId;
+    this.stateColors = {};
+    this.chart = undefined;
+    this.initStates();
+  }
+
+  /**
+   * Init the eventhandlers on the elements
+   */
+  public registerEventHandlers() {
+    this.fetchState();
+  }
+
+  /**
+   * Check if chart is loaded
+   */
+  public isLoaded(): boolean
+  {
+    return this.chart != undefined;
+  }
+
+  private fetchState() {
+    const self = this;
```

Comment:
Adding a loading spinner would be a nice addition.

#nicetohave

Reply:
Yes, I thought of that myself I didn't want to put to much effort in that.

Reply:
Added this as a feature request of https://www.pivotaltracker.com/story/show/162082458